src/main/java/org/editorconfig/core/EditorConfig.java

```diff
@@ -115,6 +125,15 @@ public EditorConfig(String configFilename, String version) {
     return result;
   }
 
+  private String getFileExtension(File file) {
+    String name = file.getName();
+    int lastIndexOf = name.lastIndexOf(".");
+    if (lastIndexOf == -1) {
```

Comment:
I prefer:

```java
if (lastIndexOf > -1) {
  return name.substring(lastIndexOf);
} else {
  return ""; // empty extension
}
```

over:

```java
if (lastIndexOf == -1) {
  return ""; // empty extension
}
return name.substring(lastIndexOf);
```

**Reason:** The second code sample highlights the exceptional case and the first code sample highlights the desired case.